Assets/BowAndArrowImproved/Scripts/GameManager.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using UnityEngine;
+
+public class GameManager : MonoBehaviour
+{
+
+    public static GameManager Instance;
+    public GameManagerState CurrentState { get; private set; } = GameManagerState.PreWave;
+    public static Action WaveDidStart;
+    public static Action WaveDidEnd;
+
+    [SerializeField] private PreWaveBallon preWaveBallon; 
+    
+    private void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+        }
+        else
+        {
+            Destroy(gameObject);
+        }
+    }
+    private void Start()
+    {
+        // Initialize the game, show the PreWaveBallon.
+        ShowPreWaveBalloon();
+    }
+    
+    public void StartWave()
+    {
+        WaveDidStart?.Invoke();
+        ChangeState(GameManagerState.Wave);
+    }
```

Comment:
first change the state, THEN invoke the event, right now - when someone gets the event, the state of the game manager hasn't even changed yet, this creates inconsistencies